server/routes/api/__test__/user.unit.test.js

```diff
@@ -1,7 +1,86 @@
+const User = require('../../../models/user');
+
 describe('User Routes', () => {
-  test('GET method', done => {
-    done();
+  let userID;
+  afterAll(() => {
+    if (userID) {
+      // Delete test user after create test
+      User.findByIdAndDelete({ _id: userID }).exec();
+    }
+  });
+
+  describe('POST /user', () => {
+    it('will return error without payload', async () => {
+      const res = await request(app)
+        .post('/api/user')
+        .type('form'); // send post with no payload
+      const errorObj = res.body.error.errors;
+      expect(res.statusCode).toEqual(400);
+      expect(Object.keys(errorObj)).toEqual(['email', 'password']); // should return two errors
+      expect(errorObj.email.message).toBe('You must provide an email address.');
+      expect(errorObj.password.message).toBe('You must provide a password.');
+    });
+
+    it('will return error without an email', async () => {
+      const res = await request(app)
+        .post('/api/user')
+        .type('form')
+        .send('password=test'); // send post with no email
+      const errorObj = res.body.error.errors;
+      expect(res.statusCode).toEqual(400);
+      expect(Object.keys(errorObj)).toEqual(['email']); // should return one error for email
+      expect(errorObj.email.message).toBe('You must provide an email address.');
+    });
+
+    it('return error without a password', async () => {
+      const res = await request(app)
+        .post('/api/user')
+        .type('form')
+        .send('email=example@test.com'); // send post with no password
+      const errorObj = res.body.error.errors;
+      expect(res.statusCode).toEqual(400);
+      expect(Object.keys(errorObj)).toEqual(['password']); // should return one error for password
+      expect(errorObj.password.message).toBe('You must provide a password.');
+    });
+
+    it('will return error with invalid email', async () => {
+      const res = await request(app)
+        .post('/api/user')
+        .type('form')
+        .send('email=test@test') // send post with invalid email address
+        .send('password=test');
+      const errorObj = res.body.error.errors;
+      expect(res.statusCode).toEqual(400);
+      expect(Object.keys(errorObj)).toEqual(['email']); // should return one error for email
+      expect(errorObj.email.message).toBe('Provide a proper email address.');
+    });
+
+    it('will return error with duplicate email', async () => {
+      const res = await request(app)
+        .post('/api/user')
+        .type('form')
+        .send('email=test@test.com') // send post with already used email address
+        .send('password=test');
+      const errorObj = res.body.error.errors;
+      expect(res.statusCode).toEqual(400);
+      expect(Object.keys(errorObj)).toEqual(['email']); // should return one error for email
+      expect(errorObj.email.message).toBe('Email already exists.');
+    });
+
+    it('will return user object on successful create', async () => {
```

Comment:
Let's test that the creation is successful, first, and _then_ test the duplicate email. I believe this can help avoid misleading errors if there's a problem on successful create of the user. Basically, move this test above the duplicate email test.

Reply:
Done.